adds docs about our github label conventions

indirectlylit opened the pull request:
### Summary

Adds a summary of our github label conventions. I'll also add descriptions to the labels in github

### Reviewer guidance

Make sure both the docs and the descriptions make sense

### References

fixes #3835

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

indirectlylit committed:
labels

benjaoming commented on docs/references/development_workflow.rst:
```diff
@@ -82,3 +84,78 @@ Within the learning equality repository:
+Priority
+~~~~~~~~
+
+These are used to sort issues and sometimes PRs by priority if *and only if* the item is assigned a milestone. Every issue in a milestone ought to have a priority label.
```
I got the idea that you would rather that a release maintainer assigns priority but that anyone with access can suggest the milestone?

indirectlylit replied:
The way we've been approaching it is that the product lead (@rtibbles) sets high-level goals about what should be accomplished in each milestone, and the product development manager (myself) sets short-term priorities within the milestone in order to achieve those goals and simultaneously address important bugs, tech debt, and UX improvements.

Both of these should happen based on input from the team, our users, and our contributors.

indirectlylit replied:
One thing I was thinking about was adding another dimension: 'severity'

that way we have some metadata which is decoupled from release planning

benjaoming replied:
Although not very relevant to the labels, I think it's important that the milestone is okay for anyone to set when they feel it's obvious which release a bug affects and will be fixed in. Or a PR already targeting a specific branch.

Maybe add in the description that priority is usually assigned by a core member?

indirectlylit replied:
> Maybe add in the description that priority is usually assigned by a core member?

will do

>  I think it's important that the milestone is okay for anyone to set when they feel it's obvious which release a bug affects and will be fixed in. Or a PR already targeting a specific branch.

Yeah for sure people should be setting milestones on the PRs and issues they're working on, or for clearly important or critical issues.

The main situation where we might need additional process is when adding work to a milestone that  might affecting timelines. In that case there should be a conversation to make sure we can have resources to work on it, or adjust date expectations if necessary.

benjaoming commented on docs/references/development_workflow.rst:
```diff
@@ -82,3 +84,78 @@ Within the learning equality repository:
+Tasks
+~~~~~
+
+These labels are generally used to categorize issues by the *outcome* of closing the issue. They are not necessarily mutually exclusive, but often are.
```
I have never understood these labels clearly, the description have unfortunately not cleared it up. Rather than using labels, shouldn't tasks always be expressed by the opening of new issues?

indirectlylit replied:
It's still worded poorly then. It's supposed to indicate "task type or category", and in particular what effect or outcome it will have. Does that make more sense?

indirectlylit replied:
updated title to "Task types"

benjaoming commented on docs/references/development_workflow.rst:
```diff
@@ -82,3 +84,78 @@ Within the learning equality repository:
+* **TAG: dev experience**
+* **TAG: needs design**
+* **TAG: needs details**
+* **TAG: needs more info**
```
needs details = needs more info?

indirectlylit replied:
yes good catch, thanks

benjaoming commented on docs/references/development_workflow.rst:
```diff
@@ -82,3 +84,78 @@ Within the learning equality repository:
+These tags can be applied to either issues or PRs to help with general organization
+
+* **TAG: beginner issue**
+* **TAG: dev experience**
```
I have never found the "dev experience" much useful, as I think devs generally tend to express their experience in description texts etc. Is it useful for sorting issues and PRs?

benjaoming replied:
Could go through its prior usage: https://github.com/learningequality/kolibri/pulls?utf8=%E2%9C%93&q=label%3A%22TAG%3A+dev+experience%22+is%3Aclosed+

indirectlylit replied:
A few concrete examples of improving the developer experience: improved linting, faster builds, better debugging instrumentation.

>  Is it useful for sorting issues and PRs?

that's what all labels are for!

indirectlylit committed:
updates based on feedback

indirectlylit commented:
added descriptions: https://github.com/learningequality/kolibri/labels

feedback welcome

indirectlylit committed:
updated based on feedback

indirectlylit commented:
updated, please take another look

benjaoming commented:
Great! I added a little "Priority: ..." to the `P*:` labels.

I would propose to merge `TAG: *` and `TASK: *` and reduce the number of items. I don't think they act mutually exclusive, although that would be the intention?

To replace `TASK`, we could add a PR check list item `* [ ] Follow-up issues have been opened and referenced if necessary` ?

indirectlylit commented:
Ah - you're interpreted "task" as indicating things that need to be done in order to close the issue. That wasn't the intent but I see how that could easily be the interpretation.

I'll think about it and return to it next week

benjaoming commented:
Yes, but also the other way -- I interpreted it both ways however seemed suitable for a given occasion :smile:

indirectlylit committed:
updates based on feedback

indirectlylit commented:
I've updated this PR and our github labels:

* merge 'tag' and 'task' labels into just 'tag's
* made a new 'todo' label type
* updated the PR template based on our usage of it
* updated the docs to be easier to keep up-to-date

benjaoming commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -35,20 +35,27 @@
+- [ ] If this includes an internal dependency change, a link to the diff is provided
+- [ ] Gherkin stories have been updated
+- [ ] Unit tests have been updated
+
```
:tada: 

This was also part of the discussion for @mrpau-richard 's work: #4544 - Namely, that we need a way to make the integration testing visible in PRs: Is it required? Is the sheet generated?

So, this is a great start!

benjaoming commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -35,20 +35,27 @@
+
+- [ ] Contributor has fully tested the PR manually
+- [ ] If there are any front-end changes, before/after screenshots are included
+- [ ] If this includes an internal dependency change, a link to the diff is provided
```
I think this should be moved up to the PR process section.

indirectlylit replied:
thanks good catch

benjaoming commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -35,20 +35,27 @@
+- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
+- External dependency files were updated if necessary (`yarn` and `pip`)
+- Documentation is updated
+- Contributor is in AUTHORS.rst
```
These need `[ ]` or not?

indirectlylit replied:
I removed the reviewer checkboxes because no one is using them. I think they're helpful tips, but maybe too much friction to require people to use

benjaoming commented on docs/references/development_workflow.rst:
```diff
@@ -15,6 +15,8 @@ At a high level, we follow the 'Gitflow' model. Some helpful references:
 Pull requests
 -------------
 
+Be sure to follow the instructions in the Github PR template.
```
Can you add a link to that?

indirectlylit replied:
udpdated

benjaoming commented on docs/references/development_workflow.rst:
```diff
@@ -82,3 +84,50 @@ Within the learning equality repository:
+Organizational Tags
+~~~~~~~~~~~~~~~~~~~
+
+Labels prefixed with **TAG:** are general-purpose, and are used to help organize issues and PRs.
```
Rather than sections with one paragraph each, would it be okay to have just a list with bolded items? Or a table?

indirectlylit replied:
that sounds fine to me but for this one I'll say "PR welcome" :)

benjaoming replied:
Right, I'll have a look if it turns out to have the right itch+scratch combo :smile:

benjaoming reviewed (dismissed):
Great work, few comments - otherwise feel free to merge at own call :+1:

indirectlylit committed:
fix PR template

indirectlylit committed:
add links
